Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -940,8 +940,10 @@ - (void)_setResourceLoadStatisticsTestingCallback:(void (^)(WKWebsiteDataStore *
 
 - (void)_setStorageAccessPromptQuirkForTesting:(NSString *)topFrameDomain withSubFrameDomains:(NSArray<NSString *> *)subFrameDomains withTriggerPages:(NSArray<NSString *> *)triggerPages completionHandler:(void(^)(void))completionHandler
 {
-    if (!_websiteDataStore->isPersistent())
+    if (!_websiteDataStore->isPersistent()) {
+        completionHandler();
```

Comment:
But then the function right above has the exact same issue, no? :)

Reply:
The function above is registering callback for specific operation, so the callback does not need to be invoked for setting failure.